### ISSUE TITLE
🛠 Tooling: dist/ directory includes *.test.* files

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/all-contributors-auto-action#184]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/all-contributors-auto-action/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

See file changes in #182: the `dist/` directory built by `build:release` includes artifacts from test. This doesn't hurt anything but is a little sloppy IMO.

* https://github.com/JoshuaKGoldberg/all-contributors-auto-action/blob/d956a966c3c9d11215b18eb25b4a1c315bf9b051/dist/getExistingContributors.d.ts#L1-L3
* https://github.com/JoshuaKGoldberg/all-contributors-auto-action/blob/d956a966c3c9d11215b18eb25b4a1c315bf9b051/dist/getExistingContributors.d.ts.map#L1

### Additional Info

cc @johnnyreilly - not something we need to change or account for in https://github.com/JoshuaKGoldberg/create-typescript-app/pull/1196 I don't think, but something to keep in mind maybe?

**Comment by johnnyreilly** [comment]
Yeah I handled it in Schemar by having a `test` directory: https://github.com/johnnyreilly/schemar/tree/main/test

**Comment by JoshuaKGoldberg** [comment]
Resolved by #1403. 🥳
